src/khepri_machine.erl

```diff
@@ -1263,6 +1264,19 @@ handle_aux(
     Tree = get_tree(State),
     ok = restore_projection(Projection, Tree, PathPattern),
     {no_reply, AuxState, IntState};
+handle_aux(_RaState, cast, tick, AuxState, IntState) ->
+    State = ra_aux:machine_state(IntState),
+    Ts = erlang:system_time(millisecond),
+    Dedups = get_dedups(State),
+    CanExpire = fun(_CommandRef, {_Reply, Expiry}) -> Expiry =< Ts end,
+    Effects = case khepri_utils:maps_any(CanExpire, Dedups) of
+                  true ->
+                      ExpireDedups = #expire_dedups{},
+                      [{append, ExpireDedups}];
```

Comment:
What about collecting the lists of expired command refs during this scan and passing it in the `#expire_dedups{}` command? This way the apply only has to call `maps:remove/2`.